zdm/survey.py

```diff
@@ -495,7 +498,8 @@ def __init__(self, state, survey_name:str,
         self.process_survey_file(filename, NFRB, iFRB)
         # DM EG
         self.init_DMEG(state.MW.DMhalo)
-        
+        # Zs
+        self.init_zs()
```

Comment:
I feel like there should be some option to turn this on or off. The reason is that it's possible to have missing redshifts, but for good reasons - e.g., one simply has not yet got an optical follow-up. For ASKAP, I expect this option will kill lots of real FRBs. Let's discuss this, but I don't think we want this here, operating in the background all the time